### PR TITLE
Bugz-504: more stable nametag updates

### PR DESCRIPTION
**Pull request by MiladNazeri**
Should be more stable for the on mode to prevent leftover nametags.
 Fixes https://highfidelity.atlassian.net/browse/BUGZ-504

### CODE REVIEW

File: scripts/simplifiedUI/ui/simplifiedNametag/resources/modules/nameTagListManager.js [hunk 1]
```diff
@@ -168,6 +154,18 @@ function getDistance(uuid, checkAvatar, shouldSave) {
 }
 
 
+// Quick check for distance from avatar
+function quickDistanceCheckForNonSelectedAvatars(uuid) {
+    var eye = MyAvatar.position;
```
Comment (zfox23) [inline scripts/simplifiedUI/ui/simplifiedNametag/resources/modules/nameTagListManager.js]: this isn't your eye position, so this variable name is misleading

File: scripts/simplifiedUI/ui/simplifiedNametag/resources/modules/nameTagListManager.js [hunk 2]
```diff
@@ -197,21 +195,35 @@ function toggleInterval() {
...
             .forEach(function (avatar) {
                 if (avatar) {
-                    add(avatar);
+                    var avatarDistance = quickDistanceCheckForNonSelectedAvatars(avatar);
+                    if (avatarDistance < MAX_RADIUS_IGNORE_METERS) {
+                        add(avatar);
+                    }
                 }
             });
+        alwaysOnAvatarDistanceCheck = Script.setInterval(maybeAddOrRemoveIntervalCheck, DISTANCE_CHECK_INTERVAL_MS);
```
Comment (zfox23) [inline scripts/simplifiedUI/ui/simplifiedNametag/resources/modules/nameTagListManager.js]: please call `maybeClearAlwaysOnAvatarDistanceCheck()` before this

File: scripts/simplifiedUI/ui/simplifiedNametag/resources/modules/nameTagListManager.js [hunk 3]
```diff
@@ -521,16 +538,20 @@ function removeNametag(uuid) {
...
+    if (avatarNametagMode === "alwaysOn") {
+        handleAvatarNametagMode("alwaysOn");
+    }
 
     return _this;
 }
 
 
-// Destory the manager and disconnect from username signal
+// Destory the manager
```
Comment (zfox23) [inline scripts/simplifiedUI/ui/simplifiedNametag/resources/modules/nameTagListManager.js]:
```suggestion
// Destroy the manager
```

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20025/HighFidelity-Beta-PR15792-6196c6b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20025/HighFidelity-Beta-Interface-PR15792-6196c6b.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20025/HighFidelity-Beta-PR15792-6196c6b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20025/HighFidelity-Beta-Interface-PR15792-6196c6b.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20025/HighFidelity-Beta-PR15792-6196c6b.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20025/HighFidelity-Quest-Beta-PR15792-6196c6b.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20032/HighFidelity-Beta-PR15792-61e4155.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20032/HighFidelity-Beta-PR15792-61e4155.exe

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20032/HighFidelity-Beta-Interface-PR15792-61e4155.dmg

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20032/HighFidelity-Beta-Interface-PR15792-61e4155.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20032/HighFidelity-Beta-PR15792-61e4155.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20032/HighFidelity-Quest-Beta-PR15792-61e4155.apk)

**Pull request merged**